src/cffi-clasp.lisp

```diff
@@ -137,13 +137,13 @@ SIZE-VAR is supplied, it will be bound to SIZE during BODY."
 (defun make-shareable-byte-vector (size)
   "Create a Lisp vector of SIZE bytes that can passed to
 WITH-POINTER-TO-VECTOR-DATA."
-  (make-array size :element-type '(unsigned-byte 8)))
-
-;; frgo, 2016-07-02: TODO: Implemenent!
-;; (defmacro with-pointer-to-vector-data ((ptr-var vector) &body body)
-;;   "Bind PTR-VAR to a foreign pointer to the data in VECTOR."
-;;   `(let ((,ptr-var (si:make-foreign-data-from-array ,vector)))
-;;      ,@body))
+  ;; borrowed from static.vectors
+  (core:make-static-vector '#.(upgraded-array-element-type '(unsigned-byte 8)) size))
```

Comment:
I'm unsure about this call to upgraded-array-element-type. What's it doing? Is it really possibly that some clasp platforms don't support ub8 arrays? Would it signal an error if you passed '(unsigned-byte 8) unconditionally? /cc @sionescu

Reply:
Yes, errors;
(core:make-static-vector  '(unsigned-byte 8) 12) 
->
Condition of type: SIMPLE-PROGRAM-ERROR
Handle make-static-vector :element-type (UNSIGNED-BYTE 8)

(upgraded-array-element-type '(unsigned-byte 8)) 
->
EXT:BYTE8

The code is borrowed from here (from drmeister, main author of clasp)
https://github.com/clasp-developers/static-vectors/blob/master/src/impl-clasp.lisp#L22

Reply:
https://github.com/cffi/cffi/pull/140#issuecomment-471514272